### PR TITLE
Feature/rational to polynomial

### PR DESCRIPTION
**Pull request by RotemKalisch**
Add functionality to convert an arbitrary rational matrix to a polynomial matrix

### CODE REVIEW

**Review by shacharwein — CHANGES_REQUESTED** [review]
This is great and very useful!
But I'm gonna be greedy and ask for one more thing which is appropriate in this Commit :)
an interesting point i discussed with Ido, is that if we multiply all matrices of a CMF by some expression, it wont effect the projective action of the matrices (meaning say the mobius transforms), and will keep the conservative property (as we could factor out the values the expression gets, and multiplication is abelian)
So - if we could take the LCM of all denominators of all matrices in a CMF, and multiply by that, we would convert a rational CMF to a polynomial CMF, with (upto singularities) the same limits.
Why would you do that? - well it might make the trajectory matrix calculations faster? I'm not sure, but its worth a try!
Bottom line - it would be nice to have an "as_polynomial" method for a CMF.
Thank you!!!

**Review by shacharwein — APPROVED** [review]
